src/scripts/main.js

```diff
@@ -1,3 +1,120 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  let sortDirection = true;
+  let selectedRow = null;
+
+  // Table sorting
+  table.querySelectorAll('th').forEach((header) => {
+    header.addEventListener('click', () => {
+      const index = Array.from(header.parentNode.children).indexOf(header);
+      const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+      rows.sort((a, b) => {
+        const cellA = a.children[index].textContent.trim();
+        const cellB = b.children[index].textContent.trim();
+
+        return sortDirection
+          ? cellA.localeCompare(cellB)
+          : cellB.localeCompare(cellA);
```

Comment:
The sorting logic uses `localeCompare`, which is suitable for strings. However, this may not work correctly for numeric columns like 'Age' and 'Salary'. Consider adding logic to handle numeric sorting for these columns.